src/plugins/condition/components/inspector/ConditionalStylesView.vue

```diff
@@ -1,14 +1,110 @@
+            @click="addConditionSet">
+            <span class="c-c-button__label">Use conditional styling</span>
+        </button>
+    </div>
+    <div v-else>
+        <div class="holder c-c-button-wrapper align-left">
+            <button
+                class="c-c-button c-c-button--minor add-criteria-button"
+                @click="removeConditionSet">
+                <span class="c-c-button__label">Remove conditional styling</span>
+            </button>
+        </div>
+        <ul>
+            <li v-for="conditionStyle in conditionalStyles"
+                :key="conditionStyle.conditionIdentifier.key">
+                <conditional-style :condition-identifier="conditionStyle.conditionIdentifier"
+                                   :condition-style="conditionStyle.style"/>
+            </li>
+        </ul>
+    </div>
+</div>
 </template>
 
 <script>
 
+import ConditionalStyle from "./ConditionalStyle.vue";
 export default {
     components: {
+        ConditionalStyle
     },
     inject: [
-        'openmct'
-    ]
+        'openmct',
+        'domainObject',
+        'layoutItem'
+    ],
+    data() {
+        return {
+            conditionalStyles: []
+        }
+    },
+    destroyed() {
+        if (this.styleRuleManager) {
+            this.styleRuleManager.destroy();
+        }
+    },
+    mounted() {
+        if (this.layoutItem) {
+            //TODO: Handle layout items
+        }
+        if (this.domainObject.configuration) {
+            this.defautStyle = this.domainObject.configuration.defaultStyle;
+            if (this.domainObject.configuration.conditionalStyle) {
+                this.conditionalStyles = this.domainObject.configuration.conditionalStyle.styles || [];
+            }
+        }
+    },
+    methods: {
+        addConditionSet() {
+            //TODO: this.conditionSetIdentifier will be set by the UI before calling this
+            this.initializeConditionalStyles();
+        },
+        removeConditionSet() {
+            this.conditionSetIdentifier = '';
+            this.conditionalStyles = [];
+            this.persist();
+        },
+        initializeConditionalStyles() {
+            this.openmct.objects.get(this.conditionSetIdentifier).then((conditionSetDomainObject) => {
+                conditionSetDomainObject.configuration.conditionCollection.forEach((identifier, index) => {
+                    this.conditionalStyles.push({
+                        conditionIdentifier: identifier,
+                        style: this.defautStyle
+                    });
+                });
+                this.persist({
+                    defaultStyle: this.defaultStyle,
+                    conditionSetIdentifier: this.conditionSetIdentifier,
+                    styles: this.conditionalStyles
+                });
+            });
+        },
+        findStyleByConditionId(id) {
+            for(let i=0, ii=this.conditionalStyles.length; i < ii; i++) {
```

Comment:
Can you help me to understand how this pattern is better than the more common `for (let i = 0; i < array.length; i++)`? Is it more performant?

Reply:
I also do not understand this :)

Reply:
Done.

Reply:
Just realized these are functionally different in the case where the array length changes during execution of the loop. Both fail, but in subtly different ways :) Modifying an array while you're looping over it is fraught with danger anyway though, and best avoided.

But yes, I also wondered if this was a style thing, or if there was a performance advantage to having a local copy of `conditionalStyles.length`?

Reply:
There is a small performance advantage if you're getting the length of properties of a nested object or a multi dimensional array - I will chalk it down to habit.
But that's ok. I'll change the habit for now :)

Reply:
Looks like there might be some very modest performance advantage in some browsers - https://jsperf.com/caching-array-length/4 . In FF I actually see worse performance though? I theorize that there used to be a more significant performance advantage which has largely disappeared with JIT compiler optimizations in modern JS engines.